### PR TITLE
fix(deprecation): #2343 Check for deprecated APIs

### PR DESCRIPTION
**Pull request by deeptibaghel**
fix(deprecation): #2343 Check for deprecated APIs

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,7 +1,18 @@
+extends:
+  - plugin:fxa/server
+  - plugin:node/recommended
```
Comment (vladikoff) [inline .eslintrc]: @deeptibaghel we would need to add this here: https://github.com/mozilla/eslint-plugin-fxa/issues/9 then `plugin:fxa/server` should have the new rules automatically and can be reused in other repositories.

File: .eslintrc [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+  - node
+
+root: true
 
 rules:
-  handle-callback-err: 0
-  strict: 2
+  node/no-missing-require: off
+  node/no-unpublished-require: off
+  node/shebang: off
```
Comment (vladikoff) [inline .eslintrc]: same as comment above, let's make changes in eslint-plugin-fxa, land those as a pull request, then change this Git SHA in this pull request: https://github.com/mozilla/fxa-auth-server/blob/master/npm-shrinkwrap.json#L338

File: package.json [hunk 3]
```diff
@@ -71,6 +71,7 @@
     "otplib": "7.0.0",
     "po2json": "0.4.5",
     "poolee": "1.0.1",
+    "punycode": "^2.1.0",
```
Comment (vladikoff) [inline package.json]: remove the `^` before the `2`

Comment (deeptibaghel) [reply package.json]: do we need to change  this as well , if yes what all we need to do ?
https://github.com/mozilla/fxa-auth-server/blob/86de08ba33540a1720cc9974b03e189394fa9105/npm-shrinkwrap.json#L81

"punycode": {
                  "version": "1.4.1",
                  "from": "https://registry.npmjs.org/punycode/-/punycode-1.4.1.tgz",
                  "resolved": "https://registry.npmjs.org/punycode/-/punycode-1.4.1.tgz"
                }

Comment (vladikoff) [reply package.json]: would have to run `npm run shrink` for that to regenerate that file

**Comment by deeptibaghel** [comment]
@vladikoff when i run "npm run shrink", it creates a file which is having just around 5000 lines instead of current 10000 lines
when i run "npm shrink --dev" it creates a file which is having around 12000 lines.
So i am bit confused now.
Also why travis is failing for node 8 is not clear to me :(

**Comment by deeptibaghel** [comment]
@vladikoff , punycode.js is done.

**Comment by vladikoff** [comment]
@pdehaan @deeptibaghel  does this work for you?

I Get:

```
 ᐅ grunt lint
Running "eslint:app" (eslint) task
Warning: Failed to load plugin node: Cannot find module 'eslint-plugin-node'
Referenced from: plugin:fxa/server
Referenced from: /Users/vladikoff/mozilla/fxa-local-dev/fxa-auth-server/.eslintrc Use --force to continue.
```

**Comment by vladikoff** [comment]
I wonder if this is because of npm2 vs npm3-5

**Comment by deeptibaghel** [comment]
> I wonder if this is because of npm2 vs npm3-5

@vladikoff yeah i am using npm3

**Comment by vladikoff** [comment]
@deeptibaghel Thank you!!! 🎁

**Pull request merged**